components/claim/components/email_to_ldap.jsx

```diff
@@ -63,19 +69,21 @@ export default class EmailToLDAP extends React.Component {
         state.ldapPassword = ldapPassword;
         this.setState(state);
 
-        checkMfa(
-            this.props.email,
-            (requiresMfa) => {
-                if (requiresMfa) {
-                    this.setState({showMfa: true});
-                } else {
-                    this.submit(this.props.email, password, '', ldapId, ldapPassword);
-                }
-            },
-            (err) => {
-                this.setState({error: err.message});
+        this.props.checkMfa(this.props.email).then((result) => {
```

Comment:
Non-blocking, but I still think this would look much nicer using `await` rather than having to nest what are essentially callbacks

Reply:
I can take a look at adding that. I was trying to avoid the proliferation of `async` functions that I talked about before, but I've since realized that making the method async instead of just firing off a Promise without returning it actually lets you test them better.